Model/Sync/Category/Processor/ProcessByProduct.php

```diff
@@ -317,4 +299,65 @@ public function unAssignProducts(array $catUnmap, int $productId, array $existin
             $this->unAssignProductFromCollection($yotpoId, $productId);
         }
     }
+
+    /**
+     * @param array<mixed> $collectionData
+     * @param int $categoryId
+     * @param array<mixed> $categories
+     * @return mixed|string
+     * @throws NoSuchEntityException
+     * @throws \Magento\Framework\Exception\LocalizedException
+     */
+    public function createOrFetchCollection($collectionData, $categoryId, $categories)
+    {
+        $yotpoIdToReturn = '';
+        $currentTime = date('Y-m-d H:i:s');
+        $url = $this->config->getEndpoint('collections');
+        $createCollection = $this->yotpoCoreApiSync->sync(
+            Request::HTTP_METHOD_POST,
+            $url,
+            $collectionData
+        );
+        if (!$createCollection) {
+            return $yotpoIdToReturn;
+        }
+        $newCollections = null;
+        $categoryIdToUpdate = null;
+        $response = $createCollection->getData('response');
+        if ($this->checkForCollectionExistsError($createCollection)) {
+            $existColls = [$categoryId];
+            $existingCollections = $this->getExistingCollectionIds($existColls);
+            $yotpoIdToReturn = is_array($existingCollections) && array_key_exists($categoryId, $existingCollections) ?
+                $existingCollections[$categoryId] :
+                null ;
+            if ($yotpoIdToReturn) {
+                $newCollections = [];
+                $newCollections[$categoryId] = [
+                    'yotpo_id' => $yotpoIdToReturn,
+                    'synced_to_yotpo' => $currentTime
+                ];
+                $categoryIdToUpdate   =   $categories[$categoryId]->getRowId()
+                    ?: $categories[$categoryId]->getId();
+            }
```

Comment:
One thing is missing over here - In case the collection name in Magento and collection name in Yotpo are different, execute a PATCH request to update the collection name in Yotpo according to the name in Magento.

That's a potential bug I wanna close, as in this case, we'll never know that the name of the collection was updated.

Seems like we don't have such method (for updating existing category) in `ProcessByProduct`, by we do have in `ProcessByCategory`, which is called `syncExistingCollection`.
Can we extract it to a shared class in the Category entity that will be used by both `ProcessByProduct` and `ProcessByCategory`?